src/ghost_borrow_mut.rs

```diff
@@ -149,7 +149,7 @@ impl<'a, 'brand, T, const N: usize> GhostBorrowMut<'a, 'brand> for &'a [GhostCel
 }
 
 
-impl<'a, 'brand, T, const N: usize> GhostBorrowMut<'a, 'brand> for [&'a GhostCell<'brand, T>; N] {
+impl<'a, 'brand, T: ?Sized, const N: usize> GhostBorrowMut<'a, 'brand> for [&'a GhostCell<'brand, T>; N] {
```

Comment:
I'm not sure this is safe, at least the comment above the `check_distinct` call says
> We require that the types are `Sized`, so no fat pointer problems.

Reply:
Thanks for bringing this up, and I realized I overlooked this when merging the previous PR, which added ?Sized to tuples of references.

First of all, I want to note that I am not sure that there's any problem with fat pointers themselves, it's just that not considering them made things simpler.

I do believe however that the current implementation may simply not work. For example, I'd expect that `[&[u8]; 4]` with 4 distinct slices of size 3 would be interpreted as `[*const (); 4]` with values `[<pointer>, <3>, <pointer>, <3>]` (or the reverse, depending on ABI):

- The last 2 slices would be ignored, which is unsafe (see the "have the same size" part of the comment).
- The 2 matching sizes would lead to `check_distinct` firing off, which is incorrect if the pointers are distinct.

I believe we would need the pointer meta-data API (currently unstable) to reliably extract the data pointer of a possibly fat pointer and check an array of _those_.

Apart from this implementation detail, I don't inherently see any unsafety there. For example, I don't believe it is possible for 2 `GhostCell<'a, [T]>` to refer to overlapping slices because obtaining the overlapping slices would require obtaining overlapping `&'a mut [T]` and building `GhostCell` of those, and the former is not possible in safe Rust.

(If it were possible to have overlapping slices, it would likely be possible to have overlapping objects)